tests/core/test_users.py

```diff
@@ -0,0 +1,25 @@
+from rancher import ApiError
+
+
+def test_user_cant_delete_self(admin_mc):
+    client = admin_mc.client
+
+    try:
+        client.delete(admin_mc.user)
+        deleted = True
+    except ApiError:
+        deleted = False
+
+    assert deleted is False
```

Comment:
instead of this try, expect, assert paradigm, can you use pytest's
```
    with pytest.raises(ApiError):
```
functionality?

Reply:
Yup. I also changed the error code returned to a 422, as it was 500ing.